### PR TITLE
Update driver

### PR DESCRIPTION
**Pull request by wep21**
This PR should be merged after real sensor testing.

### CODE REVIEW

File: livox_ros2_driver/livox_ros2_driver/lddc.cpp [hunk 1]
```diff
@@ -668,4 +672,366 @@ void Lddc::PrepareExit(void) {
...
+  using std::chrono_literals::operator""s;
+
+  bool check_pps_signal =
+    cur_node_->declare_parameter("check_pps_signal", false);
+
+  updater_.add("livox_tempereature", this, &Lddc::checkTemperature);
+  updater_.add("livox_internal_voltage", this, &Lddc::checkVoltage);
+  updater_.add("livox_motor_status", this, &Lddc::checkMotor);
+  updater_.add("livox_optical_window", this, &Lddc::checkDirty);
+  updater_.add("livox_firmware_staus", this, &Lddc::checkFirmware);
```
Comment (kenji-miyake) [inline livox_ros2_driver/livox_ros2_driver/lddc.cpp]: Please change `staus` to `status`? (This is an original typo) :pray:

File: livox_ros2_driver/livox_ros2_driver/lddc.cpp [hunk 2]
```diff
@@ -668,4 +672,366 @@ void Lddc::PrepareExit(void) {
...
 }
 
+void Lddc::initializeDiagnostics()
+{
+  using std::chrono_literals::operator""s;
+
+  bool check_pps_signal =
+    cur_node_->declare_parameter("check_pps_signal", false);
+
+  updater_.add("livox_tempereature", this, &Lddc::checkTemperature);
```
Comment (kenji-miyake) [inline livox_ros2_driver/livox_ros2_driver/lddc.cpp]: Please change `tempereature` to `temperature`? (This is an original typo) :pray:

**Comment by wep21** [comment]
`ros2 topic echo /diagnostics | grep 'hardware_id: livox' -A 10`
```
--
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_internal_voltage'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_motor_status'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_optical_window'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_firmware_status'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_service_life'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_fan_status'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_ptp_signal'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
- level: "\0"
  name: 'livox_driver: livox_time_sync'
  message: OK
  hardware_id: livox
  values:
  - key: broadcast code
    value: 3WEDH5900102431
  - key: status
    value: OK
```

**Review by kenji-miyake — APPROVED** [review]
LGTM, and the diff is clean so I think we can send a PR to upstream. @ito-san Let's consider.

**Comment by kenji-miyake** [comment]
@wep21 Can I split this commit into several ones to make sending PRs easier?

**Comment by wep21** [comment]
@kenji-miyake I created 3 branches from this PR.
https://github.com/tier4/livox_ros2_driver/tree/feature/diagnostics
https://github.com/tier4/livox_ros2_driver/tree/fix/imu-frame-id
https://github.com/tier4/livox_ros2_driver/tree/feature/use-sensor-data-qos

**Comment by kenji-miyake** [comment]
@wep21 Thank you!! :+1:

**Comment by taichiH** [comment]
@wep21 Could you devied this PR into three PRs?

**Comment by kenji-miyake** [comment]
@taichiH I'm sorry, I talked to @wep21 yesterday and I'll do it now.

**Comment by kenji-miyake** [comment]
Created https://github.com/tier4/livox_ros2_driver/pull/2 and https://github.com/tier4/livox_ros2_driver/pull/3.
`feature/use-sensor-data-qos` was already included in `tier4/master`.

**Pull request closed without merging**